flight/Modules/OnScreenDisplay/onscreendisplay.c

```diff
@@ -731,14 +731,18 @@ void draw_alarms(int x, int y, int xs, int ys, int va, int ha, int flags, int fo
 		buf[strlen(boot_reason)] = '\0';
```

Comment:
strncpy won't put a null char at the end if AlarmBootReason returns a string longer than buf so this isn't super safe.